CHANGELOG.md

```diff
@@ -3,6 +3,65 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 7.8.0
+
+_Jun 28, 2024_
+
+We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:
+
+- 🛰 Introduce server-side data source for improved server integration in the Data Grid. See [documentation](https://mui.com/x/react-data-grid/server-side-data/) for more details.
```

Comment:
Not sure if it's too descriptive, thoughts?
```suggestion
- 🛰 Introduce server-side data source for improved server integration in the Data Grid. 
  
  It supports server-side pagination, sorting, and filtering on plain and tree data along with automatic caching support.
  To enable the server-side data source, provide the `getRows` function to the `unstable_dataSource` prop on the Data Grid component which provides all the necesary information for the server to return the required data.

  ```tsx
  const dataSource = {
    getRows: async (params: GridServerGetRowsParams) => {
      const data = await fetch(
        `https://api.example.com/data?${new URLSearchParams({
          page: params.page,
          pageSize: params.pageSize,
          sortModel: JSON.stringify(params.sortModel),
          filterModel: JSON.stringify(params.filterModel),
        }).toString()}`,
      );
      return {
        rows: data.rows,
        totalRows: data.totalRows,
      };
    },

  }
  <DataGridPro
    unstable_dataSource={dataSource}
    {...otherProps}
  />
  ```

  See [plain data](https://mui.com/x/react-data-grid/server-side-data/) or [tree-data](https://mui.com/x/react-data-grid/server-side-data/tree-data/) documentation sections for more details.

```

Reply:
Shortened it slightly, let me know what you think https://github.com/mui/mui-x/pull/13655/commits/0192e3ea6156c89b3bb629481897309777201dec